storybook/src/stories/Forms/CloudDataBaseForm.stories.js

```diff
@@ -0,0 +1,165 @@
+        "An array of strings containing the names of resource groups to select",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    encryptionKeys: {
+      description: "An array of string encryption key names",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    invalidCallback: {
+      description:
+        "Function that determines invalid state and invalid text for `name` field",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    invalidTextCallback: {
+      description:
+        "A function to determine the invalid text displayed to the user and returns the string to display",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    invalidCpuCallback: {
+      description:
+        "Function that determines invalid state and invalid text for `cpu` field",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    invalidCpuTextCallback: {
+      description:
+        "A function to determine the invalid text displayed to the user and returns the string to display",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+  },
+  parameters: {
+    docs: {
+      description: {
+        component:
+          "Cloud Database Form is a form component that provides functionality for editing a Cloud Database instance.",
+      },
+    },
+  },
+};
+
+const CloudDatabaseFormStory = () => {
+  function validName(str) {
+    // regex name validation that only allows alphanumerical characters and "-", string cannot end with "-"
+    const regex = /^[A-z]([a-z0-9-]*[a-z0-9])?$/i;
+    if (str) return str.match(regex) !== null;
+    else return false;
+  }
+
+  function invalidCallback(stateData, componentProps) {
+    return (
+      !validName(stateData.name) || contains(["foo", "bar"], stateData.name)
+    );
+  }
+
+  function invalidTextCallback(stateData, componentProps) {
+    return `Invalid Name. Must match the regular expression: /^[A-z]([a-z0-9-]*[a-z0-9])?$/i`;
+  }
+
+  function invalidCpuCallback(stateData, componentProps) {
+    return (
+        !isNullOrEmptyString(stateData.cpu) &&
+        (!isWholeNumber(Number(stateData.cpu)) || 
+        (Number(stateData.cpu) !== 0 && Number(stateData.cpu) < componentProps.cpuMin) ||
+        Number(stateData.cpu) > componentProps.cpuMax)
+    );
+  }
+
+  function invalidCpuTextCallback(stateData, componentProps) {
+    return `Using dedicated cores requires a minimum of ${componentProps.cpuMin} cores and a maximum of ${componentProps.cpuMax} cores per member. For shared CPU, select 0 cores.`;
+  }
+
+  return (
+    <CloudDatabaseForm
+      data={{
+        name: "dev",
+        resource_group: "service-rg",
+        use_data: false,
+        plan: "standard",
+        encryption_key: "",
+        service: "",
+        group_id: "member",
+        memory: "",
```

Comment:
getting a warning in the console bc memory, disk, and cpu are supposed to be numbers, not strings.